application/src/main/java/org/togetherjava/tjbot/features/help/PinAnswerCommand.java

```diff
@@ -0,0 +1,81 @@
+package org.togetherjava.tjbot.features.help;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.concrete.ForumChannel;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.config.HelpSystemConfig;
+import org.togetherjava.tjbot.features.BotCommandAdapter;
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.MessageContextCommand;
+
+public final class PinAnswerCommand extends BotCommandAdapter implements MessageContextCommand {
+    private static final String COMMAND_NAME = "pin-answer";
+    private static final int MAX_PINNED_ANSWERS = 10;
+    private final String helpForumPattern;
+
+    public PinAnswerCommand(Config config) {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+        HelpSystemConfig helpConfig = config.getHelpSystem();
+        helpForumPattern = helpConfig.getHelpForumPattern();
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        Message originalMessage = event.getTarget();
+        User commandInvoker = event.getUser();
+
+        if (!(originalMessage.getChannel() instanceof ThreadChannel threadChannel)) {
+            replyNotInThread(event);
+            return;
+        }
+
+        if (!(threadChannel.getParentChannel() instanceof ForumChannel forumChannel)
+                || !forumChannel.getName().equalsIgnoreCase(helpForumPattern)) {
```

Comment:
can this all not be merged into one single if

Reply:
Right, good catch on that